src/components/App.tsx

```diff
@@ -270,6 +274,111 @@ export type ExcalidrawImperativeAPI = {
   ready: true;
 };
 
+interface ToastMessage {
+  id: string;
+  message: string;
+  delay: number;
+  autohide?: boolean;
+}
+
+export const toastState = createState({
+  data: {
+    toasts: {} as Record<string, ToastMessage>,
+  },
+  on: {
+    SHOWED_TOAST: "addToast",
+    DISMISSED_TOAST: "deleteToast",
+  },
+  actions: {
+    addToast(
+      data,
+      payload: { message: string; autohide?: boolean; delay?: number },
+    ) {
+      const id = "unique";
+      const { message, autohide = true, delay = 5000 } = payload;
+      data.toasts[id] = { id, message, autohide, delay };
+    },
+    deleteToast(data, payload: { id: string }) {
+      const { id } = payload;
+      delete data.toasts[id];
+    },
+  },
+});
+
+interface MessageProps {
+  toast: ToastMessage;
+}
+
+function Toast({ toast }: MessageProps) {
```

Comment:
use arrow functions in general
```suggestion
const Toast = ({ toast }: MessageProps) => {
```